### PR TITLE
chore: remove sfdx from local deps

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/KIT-2995

### CODE REVIEW

**Comment by louis-bompart** [comment]
Will need to admin merge to bypass lockjson validator I think

**Review by olamothe — APPROVED** [review]
As long as Salesforce team understand why were doing this change and are on board, all good.

Should we adjust the README for quantic to explain that sfdx need to be installed globally ?

**Review by fbeaudoincoveo — APPROVED** [review]

**Comment by mmitiche** [comment]
> As long as Salesforce team understand why were doing this change and are on board, all good.
> 
> Should we adjust the README for quantic to explain that sfdx need to be installed globally ?

That change looks good to me 👌 however I''m not aware of the reason behind it, can you please elaborate?

File: .github/actions/setup-sfdx/action.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: 'Setup SFDX'
+description: 'Setup SFDX'
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      run: npm i -g @salesforce/cli@2.27.6
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply .github/actions/setup-sfdx/action.yml:7]: `@latest` makes us 'lose' control of the known-last working version of SFDX.
Non-withstanding if bugs are released or not in SFDX, if there were a major release, we would update to it without control.

You can see this action.yml as a package.json in disguise in a sense 😅

**Comment by louis-bompart** [comment]
> > As long as Salesforce team understand why were doing this change and are on board, all good.
> > Should we adjust the README for quantic to explain that sfdx need to be installed globally ?
> 
> That change looks good to me 👌 however I''m not aware of the reason behind it, can you please elaborate?

Essentially, Salesforce says "Thee Shall Not install SFDX as a dependency" https://github.com/forcedotcom/cli/issues/2565#issuecomment-1815228494

I don't agree with their statement and IMHO it's a poor decision for their users, but we don't have a choice.

Frequently the package-lock.json related to SFDX will change on a whim, which creates noises. 
Moreover, the npm install fails sometimes because of this.

Finally, ideally we should have 0 isolated install, so this is a step in the right direction

<details><summary>Why no isolated installs?</summary>Isolated installs creates a new dependency tree to maintain and duplicates some package installs. This means two things essentially:

1. We have to maintain deps parity manually
2. double node_modules blackholes

</details>

**Review by erocheleau — APPROVED** [review]
Just 1 small typo to fix see suggestion

Otherwise it's good with me. ty for doing that for us!

File: packages/quantic/README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 [Salesforce cli](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli.htm#sfdx_setup_install_cli_npm)
 
-- Run `npm install --global @salesforce/cli`
+- Run `npm install --global @salesforce/cli@2.27.6`
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/quantic/README.md:20]: To be safe, 
```suggestion
- Run `npm install --global @salesforce/cli@2.x`
```

[the remark this answers is not in the document]
Comment (erocheleau) [reply packages/quantic/README.md]: Yes that's perfect IMO!

File: .github/actions/e2e-quantic-setup/action.yml [hunk 3]
```diff
@@ -13,21 +13,15 @@ runs:
...
       shell: bash
     - run: |
-        echo "${{ env.SFDX_AUTH_JWT_KEY }}" > server.key
-        ./isolated-sfdx/node_modules/.bin/sf org login jwt \
-          --client-id ${{ env.SFDX_AUTH_CLIENT_ID }} \
-          --jwt-key-file server.key \
-          --username ${{ env.SFDX_AUTH_JWT_USERNAME }} \
-          --instance-url ${{ env.SFDX_AUTH_JWT_INSTANCE_URL }} \
-          --set-default-dev-hub
+        echo "${{ inputs.SFDX_AUTH_JWT_KEY }}" > ${{ env.SFDX_AUTH_JWT_KEY_FILE }}
```
Comment (louis-bompart) [inline .github/actions/e2e-quantic-setup/action.yml:16]:
```suggestion
        echo "${{ inputs.jwtkey }}" > ${{ env.SFDX_AUTH_JWT_KEY_FILE }}
```

**Pull request merged**